### PR TITLE
test: move all generic page tests to jest

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

**Comment by JoelEinbinder** [comment]
interception.jest looks broken, missing HEADLESS

File: test/interception.jest.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ const fs = require('fs');
 const path = require('path');
 const { helper } = require('../lib/helper');
 const vm = require('vm');
-const {FFOX, CHROMIUM, WEBKIT} = require('./utils').testOptions(browserType);
+const {FFOX, CHROMIUM, WEBKIT} = testOptions;
```
Comment (JoelEinbinder) [inline test/interception.jest.js:22]: I wonder how it worked before? HEADLESS was in global somewhere?
```suggestion
const {FFOX, CHROMIUM, WEBKIT, HEADLESS} = testOptions;
```

File: test/jest/fixtures.js [hunk 2]
```diff
@@ -63,6 +63,7 @@ module.exports = function registerFixtures(global) {
   });
 
   global.registerWorkerFixture('playwright', async({}, test) => {
+    Error.stackTraceLimit = 15;
```
Comment (JoelEinbinder) [inline test/jest/fixtures.js:66]: why is this needed? Seems like a strange place to put it.

File: test/jest/fixtures.js [hunk 3]
```diff
@@ -122,14 +123,24 @@ module.exports = function registerFixtures(global) {
   
   global.registerWorkerFixture('browser', async ({browserType, defaultBrowserOptions}, test) => {
     const browser = await browserType.launch(defaultBrowserOptions);
-    await test(browser);
-    await browser.close();
+    try {
+      await test(browser);
```
Comment (JoelEinbinder) [inline test/jest/fixtures.js:127]: test can throw? I thought it catches internally?

**Review by JoelEinbinder — APPROVED** [review]

File: test/interception.jest.js (hunk 1 above)
Comment (pavelfeldman) [reply test/interception.jest.js:22]: Yes, I think dgozman injected it into the environment

File: test/jest/fixtures.js (hunk 2 above)
Comment (pavelfeldman) [reply test/jest/fixtures.js:66]: Our stacks are long and some tests match file names in them. We have it in test.js, I need it in the vm, hence the strange place. Will figure it out!

File: test/jest/fixtures.js (hunk 3 above)
Comment (pavelfeldman) [reply test/jest/fixtures.js:127]: I never made it catch and now I can use it...
